src/utils/collections.zig

```diff
@@ -64,6 +64,188 @@ pub fn RecyclingList(
     };
 }
 
+pub fn SplitUnionList(TaggedUnion: type) type {
```

Comment:
would be good to add a doc comment explaining the high level of this helper

Reply:
f70e4d5f12d6c217a98b11584c006902112d329c